test/integration/sb2-import-extension-monitors.js

```diff
@@ -0,0 +1,107 @@
+    vm.attachStorage(makeTestStorage());
+
+    // Start VM, load project, and run
+    vm.start();
+    vm.clear();
+    vm.setCompatibilityMode(false);
+    vm.setTurboMode(false);
+    vm.loadProject(invisibleVideoMonitorProject).then(() => {
+        t.equal(vm.extensionManager.isExtensionLoaded('videoSensing'), false);
+        t.equal(vm.runtime._monitorState.size, 0);
+        t.end();
+    });
+});
+
+test('loading sb2 project with visible video monitor should not load extension', t => {
+    const vm = new VirtualMachine();
+    vm.attachStorage(makeTestStorage());
+
+    // Start VM, load project, and run
+    vm.start();
+    vm.clear();
+    vm.setCompatibilityMode(false);
+    vm.setTurboMode(false);
+    vm.loadProject(visibleVideoMonitorProject).then(() => {
+        t.equal(vm.extensionManager.isExtensionLoaded('videoSensing'), false);
+        t.equal(vm.runtime._monitorState.size, 0);
+        t.end();
+    });
+});
+
+// This test looks a little different than the rest because loading a project with
+// the video sensing block requires a mock renderer and other setup, so instead
+// we are just using deserialize to test what we need instead
+test('sb2 project with video sensing blocks and monitor should load extension but not monitor', t => {
+    const vm = new VirtualMachine();
+
+    sb2.deserialize(visibleVideoMonitorAndBlocksProject, vm.runtime).then(project => {
+        // Extension loads but monitor does not
+        project.extensions.extensionIDs.has('videoSensing');
+        // Non-core extension monitors haven't been added to the runtime
+        t.equal(vm.runtime._monitorState.size, 0);
+        t.end();
+    });
+});
+
+test('sb2 project with invisible music monitor should not load monitor or extension', t => {
+    const vm = new VirtualMachine();
+    vm.attachStorage(makeTestStorage());
+
+    // Start VM, load project, and run
+    vm.start();
+    vm.clear();
+    vm.setCompatibilityMode(false);
+    vm.setTurboMode(false);
+    vm.loadProject(invisibleTempoMonitorProject).then(() => {
+        t.equal(vm.extensionManager.isExtensionLoaded('music'), false);
+        t.equal(vm.runtime._monitorState.size, 0);
+        t.end();
+    });
+});
+
+test('sb2 project with visible music monitor should load monitor and extension', t => {
+    const vm = new VirtualMachine();
+    vm.attachStorage(makeTestStorage());
+
+    // Start VM, load project, and run
+    vm.start();
+    vm.clear();
+    vm.setCompatibilityMode(false);
+    vm.setTurboMode(false);
+    vm.loadProject(visibleTempoMonitorProject).then(() => {
+        t.equal(vm.extensionManager.isExtensionLoaded('music'), true);
+        t.equal(vm.runtime._monitorState.size, 1);
+        t.equal(vm.runtime._monitorState.has('music_getTempo'), true);
+        t.equal(vm.runtime._monitorState.get('music_getTempo').visible, true);
+        t.end();
+        process.nextTick(process.exit); // This is needed because this is the end of the last test in this file!!
```

Comment:
Just FYI you can replace this with `tap.tearDown(() => process.nextTick(process.exit));` at the top of this file, to the same effect but without the test order dependency.